src/overlays/actors/ovl_En_Ru/z_en_ru.c

```diff
@@ -76,34 +86,285 @@ static DamageTable D_80A39488 = {
+        Math_SmoothStepToS(&this->torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+
+    EnRu_UpdateEyes(this, 3);
+    EnRu_PlayWalkingSound(this, play);
+    SubS_FillLimbRotTables(play, this->limbRotTableY, this->limbRotTableZ, RU2_LIMB_MAX);
+}
+
+void EnRu_DoNothing(EnRu* this, PlayState* play) {
+}
+
+void EnRu_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRu* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gAdultRutoSkel, NULL, this->jointTable, this->morphTable, RU2_LIMB_MAX);
+    EnRu_ChangeAnimation(&this->skelAnime, 0);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    // path is prepared but unused after this
+    this->path = SubS_GetPathByIndex(play, RU_GET_PATH(thisx), 0x3F);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnRu_DoNothing;
+    this->actor.gravity = -4.0f;
+}
+
+void EnRu_Destroy(Actor* thisx, PlayState* play) {
+    EnRu* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnRu_Update(Actor* thisx, PlayState* play) {
+    EnRu* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+    EnRu_UpdateModel(this, play);
+    EnRu_CollisionCheck(this, play);
+}
+
+s32 EnRu_OverrideLimbdraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx,
+                          Gfx** gfx) {
+    EnRu* this = THIS;
+
+    if (limbIndex == RU2_LIMB_HEAD) {
+        Matrix_Translate(1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        Matrix_RotateXS(this->headRot.y, MTXMODE_APPLY);
+        Matrix_RotateZS(this->headRot.x, MTXMODE_APPLY);
+        Matrix_Translate(-1500.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
+
+    if (limbIndex == RU2_LIMB_TORSO) {
+        Matrix_RotateYS(this->torsoRot.y, MTXMODE_APPLY);
+        Matrix_RotateXS(this->torsoRot.x, MTXMODE_APPLY);
+    }
+
+    if ((limbIndex == RU2_LIMB_TORSO) || (limbIndex == RU2_LIMB_LEFT_UPPER_ARM) ||
+        (limbIndex == RU2_LIMB_RIGHT_UPPER_ARM)) {
+        rot->y += (s32)(s16)(Math_SinS(this->limbRotTableY[limbIndex]) * 200.0f);
+        rot->z += (s32)(s16)(Math_CosS(this->limbRotTableZ[limbIndex]) * 200.0f);
+    }
+
+    return false;
+}
+
+void EnRu_PostLimbdraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
+    EnRu* this = THIS;
+    Vec3f headFocus = sHeadFocusPos;
+    Vec3f bodyPartPos = sBodyFocusPos;
+
+    if (sBodyPartPosIndices[limbIndex] >= 0) {
+        Matrix_MultVec3f(&bodyPartPos, &this->bodyPartsPos[sBodyPartPosIndices[limbIndex]]);
+    }
+    if (limbIndex == RU2_LIMB_HEAD) {
+        Matrix_MultVec3f(&headFocus, &thisx->focus.pos);
+    }
+
+    // pos not rot for some reason? not sure what this is doing
```

Comment:
Firstly, it is doing this for all of them, so putting the comment in the middle is misleading. Secondly, it's translating and rotating the base position (which is what the data is) using the current matrix, which for the head places the focus in the right place, for the limbs with a corresponding bodyPartPos, it saves the limb's position into that; you may find it helpful to remember that the original name for this function was `Matrix_Position`.